tests/datasets/test_oscd.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import glob
+import os
+import shutil
+from pathlib import Path
+from typing import Generator
+
+import pytest
+import torch
+import torch.nn as nn
+from _pytest.monkeypatch import MonkeyPatch
+from torch.utils.data import ConcatDataset
+
+import torchgeo.datasets.utils
+from torchgeo.datasets import OSCD
+
+
+def download_url(url: str, root: str, *args: str, **kwargs: str) -> None:
+    shutil.copy(url, root)
+
+
+class TestOSCD:
+    @pytest.fixture
+    def dataset(
+        self, monkeypatch: Generator[MonkeyPatch, None, None], tmp_path: Path
+    ) -> OSCD:
+        monkeypatch.setattr(  # type: ignore[attr-defined]
+            torchgeo.datasets.oscd, "download_url", download_url
+        )
+        md5 = "d6ebaae1ea0f3ae960af31531d394521"
+
+        monkeypatch.setattr(OSCD, "md5", md5)  # type: ignore[attr-defined]
+        url = os.path.join("tests", "data", "oscd", "OSCD.zip")
+        monkeypatch.setattr(OSCD, "url", url)  # type: ignore[attr-defined]
+
+        root = str(tmp_path)
+        transforms = nn.Identity()  # type: ignore[attr-defined]
+        return OSCD(root, transforms=transforms, download=True, checksum=True)
+
+    def test_getitem(self, dataset: OSCD) -> None:
+        x = dataset[0]
+        assert isinstance(x, dict)
+        assert isinstance(x["image"], torch.Tensor)
+        assert isinstance(x["mask"], torch.Tensor)
```

Comment:
You should add tests for shapes in `test_getitem` as well. To check the image shape we should do `assert x["image"].shape[:2] == (2, 13)` and `assert x["image"].ndim == 4`. Masks should only have 2 dims so we can use `assert x["mask"].ndim == 2`.